### PR TITLE
Feature/fix request queue limit

### PR DESCRIPTION
**Pull request by TimofeyBurak**

### CODE REVIEW

File: Softeq.XToolkit.HttpClient/Network/HttpRequestScheduler.cs [hunk 1]
```diff
@@ -172,7 +172,7 @@ private void AddNewTask(HttpRequestPriority priority, HttpRequestScheduledTaskBa
         {
             _priorityToTasksCountMap.TryGetValue(priority, out var itemsCount);
 
-            if (itemsCount > _config.MaxHttpLoadThreads)
```
Comment (wcoder) [inline Softeq.XToolkit.HttpClient/Network/HttpRequestScheduler.cs:175]: Looks like should be reverted because `MaxHttpLoadThreads` is right and already used for the control task queue.
What is the reason for the change?

Comment (TimofeyBurak) [reply Softeq.XToolkit.HttpClient/Network/HttpRequestScheduler.cs:175]: It was my assumption that there's a mistake here because:
- `MaxHttpLoadThreads` also used to determine the amount of background threads, which are actually executing requests;
- it's not clear why the size of the queue should be exactly the same as the amount of worker threads. especially providing that requests that exceed the limit are discarded (not cancelled and not executed, ever). why can't we keep them in queue a bit longer to wait for their turn?
- `HttpRequestPerPriorityLimit` was a part of the config, but was never used in code;
- in the code we get the amount of tasks of the provided priority, and then compare it to the "limit of http requests per priority"(`HttpRequestPerPriorityLimit`) - which looks more reasonable, than comparing it to the "amount of http load threads"('MaxHttpLoadThreads').

Comment (wcoder) [reply Softeq.XToolkit.HttpClient/Network/HttpRequestScheduler.cs:175]: I double-checked, and you are right! The property was missed.

What about renaming `HttpRequestPerPriorityLimit` property to `MaxRequestsPerPriority` to be more consistent?

Comment (TimofeyBurak) [reply Softeq.XToolkit.HttpClient/Network/HttpRequestScheduler.cs:175]: Done. I've also renamed `MaxHttpLoadThreads` to `HttpLoadThreads`, because it's not really a max amount - it's the exact amount of worker threads

**Review by yauhenipakala — APPROVED** [review]
